src/Raven.Client/Json/Converters/RavenTypeJsonConverter.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace Raven.Client.Json.Converters
+{
+    internal abstract class RavenTypeJsonConverter<T> : RavenJsonConverter
+    {
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (!(writer is BlittableJsonWriter blittableJsonWriter))
+            {
+                throw new SerializationException(
+                    $"Try to write {nameof(T)} property/field by {writer.GetType()} which is unsuitable reader. Should use {nameof(BlittableJsonWriter)}");
+            }
+
+            if (!(value is T tValue))
+            {
+                throw new SerializationException($"Try to write {value.GetType()} with {GetType().Name}. The converter should be used only for {nameof(T)}");
```

Comment:
`value` can be null here

Reply:
As I saw Newtonsoft.Json handle null by himself and doesn't  send it to converters

Reply:
done